src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/helm.py

```diff
@@ -187,12 +187,16 @@ def _get_helm_command():
             _, stderr = p.communicate()
         except OSError as inner:
             logger.debug("Could not run '%s' command. Exception: %s", helm_command, str(inner))
-            raise CLIError(helm_not_installed)
+            if is_diagnostics_context:
+                return None, HELM_COMMAND_ERROR
+            raise CLIError(HELM_COMMAND_ERROR.get_error_message())
 
     if stderr:
-        raise CLIError(stderr.decode())
+        if is_diagnostics_context:
+            return None, UNEXPECTED_ERROR.append_error_message(stderr.decode())
+        raise CLIError(UNEXPECTED_ERROR.append_error_message(stderr.decode()).get_error_message())
```

Comment:
Can we combine get_error_message and append_error_message so we only have one call here?

Reply:
I think it's hard, the idea for `append_error_message` is to return a ErrorClass model with the appended error message. We use this in other methods that deals with ErrorClass objects, not with messages, which make it difficult to combine both.

Reply:
The problem is that, in diagnostics context, we want to deal with a `ErrorClass` object, while raising a CLI error we want a string (obtained by `get_error_message`.